CONTRIBUTING.md

```diff
@@ -61,7 +61,7 @@ why we ask this as well as instructions for how to proceed, see the
   cd citus
   ./configure
   make
-  sudo make install
+  sudo make install-all
```

Comment:
All these lines need a comment to say that you should use `make install` for regular use and `make install-all`, when installing for development.

Reply:
Following commands is tempting, many people simply copy & paste the commands, ignoring what the comments tell. 

If the  goal is to pass the regression tests, can we  add an additional test file output for multi_extension which passes with `make install`?

Or, better, is there a possibility to move the related downgrade tests to a separate file (and have two output  files for it), and only that fails? I think it makes more sense to not have an additional file for multi_extension but for only the parts  that are  affected by `make install all`

Reply:
Separating downgrade tests might make more sense actually

Reply:
I'm still not on board with this change. There is a danger that comes with `install-all` and we're currently not even mentioning it. Plus, even if we mention  as Jelte noted, it is very implicit -- likely that users install with the wrong command for their purposes. 

If the problem is related with regression tests, we should probably fix that instead of documentation testing. 


If we can separate downgrade tests from multi_extension, then is it much easier to communicate even if `extension_downgrade` test fails. Plus, we can add the extra output file, which I'd be OK

Reply:
Is there any harm in installing downgrade scripts on `make install`?

It seems the main reason we don't install them is that we want to do packaging in a very specific way, but since we build the packages it seems to make a bit more sense to do the non-standard thing in the packaging pipeline than to force all people trying out open source Citus to do the non-standard thing.

Reply:
> Is there any harm in installing downgrade scripts on make install?

We'll have to communicate this to devrim too, for the pgdg packages. 

> It seems the main reason we don't install them is that we want to do packaging in a very specific way, but since we build the packages it seems to make a bit more sense to do the non-standard thing in the packaging pipeline than to force all people trying out open source Citus to do the non-standard thing.

Well, the downgrade scripts installed using `make install-all` are not really useful to people anyway. Because they won't allow you to downgrade to the version you just installed. Only to a version below. So they are also not really necessary for people to install, because they cannot use them for anything useful. The only thing they can use them for is making sure the tests pass.

Reply:
> We'll have to communicate this to devrim too, for the pgdg packages.

Is he planning to maintain downgrade packages? Seems unlikely.

> The only thing they can use them for is making sure the tests pass.

That seems pretty useful. It instils some level of confidence if I try out Citus from this repo.